src/StoreBase.ts

```diff
@@ -60,178 +56,211 @@ export abstract class StoreBase {
         assert.ok(this._triggerBlockCount >= 0, 'Over-popped trigger blocks!');
 
         if (this._triggerBlockCount === 0) {
-            // Go through the list of stores awaiting resolution and resolve them all
-            const awaitingList = this._triggerBlockedStoreList;
-            this._triggerBlockedStoreList = [];
-            _.forEach(awaitingList, store => {
-                store._resolveThrottledCallbacks();
-            });
+            StoreBase._resolveCallbacks();
         }
     }
 
     static setThrottleStatus(enabled: boolean) {
         this._bypassThrottle = !enabled;
+        
+        StoreBase._resolveCallbacks();
+    }
 
-        // If we're going to bypass the throttle, trigger all pending stores now
-        if (this._bypassThrottle) {
-            let pendingThrottledStore = this._pendingThrottledStores.shift();
-            while (!!pendingThrottledStore) {
-                pendingThrottledStore._resolveThrottledCallbacks();
-                pendingThrottledStore = this._pendingThrottledStores.shift();
-            }
-        }
+    constructor(private _throttleMs?: number, private _bypassTriggerBlocks = false) {
     }
 
-    constructor(throttleMs: number = 0, bypassTriggerBans = false) {
-        this._throttleMs = throttleMs;
-        this._bypassTriggerBlocks = bypassTriggerBans;
+    private static _updateExistingMeta(meta: CallbackMetadata | undefined, throttledUntil: number|undefined, bypassBlock: boolean) {
```

Comment:
Consider putting `_updateExistingMeta` after `trigger`, similar to `_resolveThrottleCallbacks`.